strawberry/custom_scalar.py

```diff
@@ -31,42 +33,80 @@ def identity(x):
     return x
 
 
+# TODO: Maybe rename this to StrawberryScalar?
 @dataclass
 class ScalarDefinition(StrawberryType):
-    name: str
-    description: Optional[str]
-    specified_by_url: Optional[str]
-    serialize: Optional[Callable]
-    parse_value: Optional[Callable]
-    parse_literal: Optional[Callable]
+    origin: Optional[type] = None
+    name: str = None
+    description: Optional[str] = None
+    specified_by_url: Optional[str] = None
+    serialize: Optional[Callable] = None
+    parse_value: Optional[Callable] = None
+    parse_literal: Optional[Callable] = None
     directives: Iterable[object] = ()
 
     # Optionally store the GraphQLScalarType instance so that we don't get
     # duplicates
     implementation: Optional[GraphQLScalarType] = None
 
-    def copy_with(
-        self, type_var_map: Mapping[TypeVar, Union[StrawberryType, type]]
-    ) -> Union[StrawberryType, type]:
-        return super().copy_with(type_var_map)
+    def to_graphql_core(self):
```

Comment:
can we keep this in the schema converter?

Reply:
I'll have a try.